app/services/SubscriptionDataService.scala

```
@@ -134,7 +132,7 @@ class SubscriptionDataService @Inject()(dataCacheConnector: DataCacheConnector,
         }
       }
       _ <- updatedDataResponse match {
-        case Some(x) => dataCacheConnector.clearCache().flatMap(r => Future.successful(r))
+        case Some(x) => dataCacheService.clearCache().flatMap(r => Future.successful(r))
         case None => Future.successful(None)
       }
     } yield {
```

This for comprehension appears to swallow/mask failures

---

I think we swallow the result type but not any errors - added a test for this